native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java

```diff
@@ -304,6 +308,34 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         }
     }
 
+    static class RecursiveDirCopyVisitor extends SimpleFileVisitor<Path> {
+
+        final Path source;
+        final Path target;
+        final CopyOption[] copyOptions;
+
+        RecursiveDirCopyVisitor(Path source, Path target, CopyOption... copyOptions) {
+            this.source = source;
+            this.target = target;
+            this.copyOptions = copyOptions;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+            Path targetPath = target.resolve(source.relativize(dir));
+            if (!Files.exists(targetPath)) {
+                Files.createDirectory(targetPath);
```

Comment:
what if parent directories do not exist. Don't we need to use `Files. createDirectories` API to create all the missing directories of the target path

Reply:
This does not create all the missing directories. If the parent exists, only a child directory is created by this. The existing API also works like this.